GraphStructure/Vertex.java

```diff
@@ -0,0 +1,16 @@
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+
+public class Vertex{
+	public Integer id;
```

Comment:
Try follow google style guide for java. Clearly it is not recommended to specify a public member. Download the style guide's xml into eclipse, it will help you format the code. However, you must be aware of the format as well. Please read the guide before coding, and align your code accordingly.